tdp/core/collections.py

```diff
@@ -137,88 +137,120 @@ def operations(self) -> dict[str, Operation]:
-
-                for operation in operations_list:
-                    if operation.name in self._dag_operations:
-                        # Merge the operation with the existing one
+    def _init_operations(
+        self, collections: Mapping[str, Collection]
+    ) -> tuple[dict[str, Operation], dict[str, Operation]]:
+        dag_operations: dict[str, Operation] = {}
+        other_operations: dict[str, Operation] = {}
+
+        for collection in collections.values():
+            # Load DAG operations from the dag files
+            for dag_file in collection.dag_yamls:
+                for read_operation in read_tdp_lib_dag_file(dag_file):
+                    existing_operation = dag_operations.get(read_operation.name)
+
+                    # The read_operation is associated with a playbook defined in the
+                    # current collection
+                    if _ := collection.playbooks.get(read_operation.name):
+                        # TODO: would be nice to dissociate the Operation class from the playbook and store the playbook in the Operation
+                        dag_operation_to_register = Operation(
+                            name=read_operation.name,
+                            collection_name=collection.name,
+                            host_names=collection.get_hosts_from_playbook(
+                                read_operation.name
+                            ),
+                            depends_on=read_operation.depends_on.copy(),
+                        )
+                        # If the operation is already registered, merge its dependencies
+                        if existing_operation:
+                            dag_operation_to_register.depends_on.extend(
+                                dag_operations[read_operation.name].depends_on
+                            )
+                            # Print a warning if we override a playbook operation
+                            if not existing_operation.noop:
+                                logger.debug(
+                                    f"'{read_operation.name}' defined in "
+                                    f"'{existing_operation.collection_name}' "
+                                    f"is overridden by '{collection.name}'"
+                                )
+                        # Register the operation
+                        dag_operations[read_operation.name] = dag_operation_to_register
+                        continue
+
+                    # The read_operation is already registered
+                    if existing_operation:
                         logger.debug(
-                            f"DAG Operation '{operation.name}' defined in collection "
-                            f"'{self._dag_operations[operation.name].collection_name}' "
-                            f"is merged with collection '{collection_name}'"
+                            f"'{read_operation.name}' defined in "
```

Comment:
shouldn't you write the playbook here where the operation is defined ?

Reply:
What do you mean?

Reply:
sorry yes it it good